lib/Sema/CSDiagnostics.cpp

```diff
@@ -6224,3 +6224,17 @@ bool CoercionAsForceCastFailure::diagnoseAsError() {
       .highlight(getSourceRange());
   return true;
 }
+
+bool KeyPathRootTypeMismatchFailure::diagnoseAsError() {
+  auto locator = getLocator();  
+  assert(locator->isKeyPathRoot() && "Expect a key path root");
+  
+  auto *subscriptExpr = castToExpr<SubscriptExpr>(getAnchor());
+  auto *tupleExpr = cast<TupleExpr>(subscriptExpr->getIndex());
+  auto keyPathType = getType(tupleExpr->getElement(0));
+  auto baseType = getType(subscriptExpr->getBase());
```

Comment:
Actually I think at least for contextual "base" vs. actual base you could use `get{From/To}Type()` right? Because that's where the mismatch is recorded?

Reply:
Yeap, we can use that :)